concordium-std/src/types.rs

```diff
@@ -581,6 +584,109 @@ pub struct ExternReturnValue {
     pub(crate) current_position: u32,
 }
 
+/// The current public balances of an account.
+#[derive(Debug, Copy, Clone)]
+pub struct AccountBalance {
+    /// The total balance of the account. Note that part of this balance might
+    /// be staked and/or locked in scheduled transfers.
+    pub(crate) total:  Amount,
+    /// The current staked amount of the account. This amount is used for
+    /// staking.
+    pub(crate) staked: Amount,
+    /// The current amount locked in releases that resulted from transfers with
+    /// schedule. A locked amount can still be used for staking.
+    pub(crate) locked: Amount,
+}
```

Comment:
What was the main reason for having these private?

I do understand that total >= staked, locked, but having these private leads to more boilerplate below which to me does not seem worthwhile.

Was there some other reason?

Reply:
The reason is to future-proof the api by not leaking the struct details.
It also makes the api for different balances more alike (here I am thinking about `available()`)

Reply:
I don't think future-proofing is a good reason. We'll have a major version release anyhow if we change the data returned by the node which is the only place this is relevant.

And if we change the data then the `new` function will in any case have to be changed, so the future proofing isn't really going to be that.

Reply:
Okay, the fields are now public and I removed the corresponding getters